### PR TITLE
fix: Objects wrongly getting converted to Array

### LINKED ISSUE #2198: Object attributes transformed in array
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
When I get an Object attribute on my object with Parse 5.2, it returns an array instead of the expected object.
(Parse 5.1 return an Object)

### Steps to reproduce
Get an Object attribute with parse 5.2

Server
- Parse Server version: `7.1`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `7`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local`

Client
- Parse JS SDK version: `5.2`

### Logs

Parse 5.1:

Parse 5.2:

### PR DESCRIPTION
**Pull request by mkmandar123**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-JS/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

Closes: https://github.com/parse-community/Parse-SDK-JS/issues/2198

## Approach
Issue demo

### CODE REVIEW

File: src/ObjectStateMutations.ts [hunk 1]
```diff
@@ -177,6 +177,12 @@ function nestedSet(obj, key, value) {
   }
 }
 
+function isJSONArray(val) {
```
Comment (dplewis) [inline src/ObjectStateMutations.ts:180]: Nice! Very clean. Can you add a comment on what a JSON Array is. You can move the one in `commitServerChange`

File: integration/test/ParseObjectTest.js [hunk 2]
```diff
@@ -2187,4 +2187,37 @@ describe('Parse Object', () => {
       Parse.allowCustomObjectId = false;
     });
   });
+
+  fit('returns correct field values', async () => {
```
Comment (dplewis) [inline integration/test/ParseObjectTest.js:2191]: Can you remove `fit`, clean up the test and add any additional values you can think of?

Also can you add a unit test to ensure the internals still work this change to JSONArray.
https://github.com/parse-community/Parse-SDK-JS/blob/alpha/src/__tests__/ObjectStateMutations-test.js#L316

**Review by dplewis — CHANGES_REQUESTED** [review]
Looks good, just needs some clean up.

File: integration/test/ParseObjectTest.js (hunk 2 above)
Comment (mtrezza) [reply integration/test/ParseObjectTest.js:2191]: Please keep `fit` until you have added and uncommented all the values from https://github.com/parse-community/Parse-SDK-JS/pull/2201 and the test passes.

**Comment by mtrezza** [comment]
I don't believe this fixes the issue, see https://github.com/parse-community/Parse-SDK-JS/issues/2198#issuecomment-2210725825. There is a fundamental question we need to solve. Let's keep discussing it in the issue thread there.

**Comment by mkmandar123** [comment]
> Looks good, just needs some clean up.

done the cleanup

**Comment by mtrezza** [comment]
Superseded by https://github.com/parse-community/Parse-SDK-JS/pull/2201; thanks for the help in the investigation @mkmandar123

**Pull request closed without merging**